pkg/reconciler/service/service_test.go

```diff
@@ -125,33 +126,33 @@ func TestReconcile(t *testing.T) {
 		WantStatusUpdates: []clientgotesting.UpdateActionImpl{{
 			Object: DefaultService("byo-rev", "foo", WithNamedRevision,
 				// The first reconciliation will initialize the status conditions.
-				WithInitSvcConditions),
+				WithInitSvcConditions, WithServiceObservedGenFailure,
```

Comment:
Do we want users to see this reason & message?

I wonder if this is surfacing bug where a condition isn't updated

Reply:
I suspect these are bugs (or places where we could be better surfacing errors back to the client). My goal with this PR is to ensure the unit tests are actually reconciling new generations (which better matches the real-world flow).

After this PR I want to just go through all the references to the ObsGenFailure and investigate a real Reason/Message for them.